### DISCUSSION TITLE
Help to setup correct Architecture

### DISCUSSION THREAD

**Discussion by vsharma009** [react-hook-form/react-hook-form#1876] [General]
Hi All,

I have a very big form which has traveller details and payment and billing details and I want to create react hook form but there is one issue that we are showing some big components between traveler and payment section so how can I configure it. i am not allowed to change UI. form should be like 
traveler section , insurance, seatmap some other ancillaries then payment details and then book button.
For live example: please check https://www.cheapoair.com and go to payment page.

Please contact me if someone can help.. main challenge is that I don't want to render middle components on form render.

**Comment by bluebill1049** [comment]
Take a look at this approach: https://www.youtube.com/watch?v=CeAkxVwsyMU which I designed for use at work.

**Reply by vsharma009** [reply]
Management is not allowing us to break form into multiple steps. we need to keep UI same and need to show only one submit button.
